### ISSUE TITLE
`-Amode=jspecify`: reports on `Void` type arguments, `assert` checks, and `@SuppressWarnings("NullAway")`

### ISSUE THREAD

**Issue by aosen-xiong** [eisop/checker-framework#2085]
Running `-Amode=jspecify` on reactor-pool and reactor-core, which already pass NullAway, gives 666 EISOP warnings. 253 of them come from the three behaviors below.

Environment: EISOP 3.49.5-eisop2-SNAPSHOT, javac 25, JSpecify 1.0.0.

**1. `Void` is treated as `@Nullable`** (104 warnings)

```java
@NullMarked
class VoidArg {
  static class Box<T> {}

  Box<Void> box;
}
```

```
VoidArg.java:7: error: [type.argument.type.incompatible] incompatible type argument for type parameter T of Box.
  Box<Void> box;
      ^
  found   : @Nullable Void
  required: @NonNull Object
```

Expected: no error. JSpecify has no special rule for `Void`, and NullAway accepts this.

**2. `assert x != null` does not refine** (110 warnings)

```java
@NullMarked
class AssertRefine {
  @Nullable String s;

  int len() {
    assert s != null;
    return s.length();
  }
}
```

```
AssertRefine.java:10: error: [dereference.of.nullable] dereference of possibly-null reference s
```

No error with `-AassumeAssertionsAreEnabled`. Both projects run NullAway with `AssertsEnabled=true`. Should `-Amode=jspecify` enable this flag, or leave it to the user?

**3. `@SuppressWarnings("NullAway")` is not honored** (39 warnings)

```java
@NullMarked
class Suppressed {
  @SuppressWarnings("NullAway")
  static String s = null;
}
```

```
Suppressed.java:6: error: [assignment.type.incompatible] incompatible types in assignment.
  found   : @Nullable NullType
  required: @NonNull String
```

Code migrating from NullAway already carries these suppressions. The count also covers reactor-core's configured alias names `DataFlowIssue` and `NotNullFieldNotInitialized`. Should the mode accept `"NullAway"` as a suppression key?
